Index: grub/grub-core/disk/efi/efidisk.c
===================================================================
--- grub.orig/grub-core/disk/efi/efidisk.c
+++ grub/grub-core/disk/efi/efidisk.c
@@ -504,6 +504,29 @@ grub_efidisk_open (const char *name, str
   if (! d)
     return grub_error (GRUB_ERR_UNKNOWN_DEVICE, "no such device");

+  /* before we open the disk, check if it still exists.
+   * efi may free the handler when device became unconnected. */
+  grub_efi_uintn_t num_handles;
+  grub_efi_handle_t *handles, *handle;
+  int found = 0;
+  handles = grub_efi_locate_handle (GRUB_EFI_BY_PROTOCOL, &block_io_guid,
+                0, &num_handles);
+  if (! handles)
+        return grub_error (GRUB_ERR_NONE, "unknown error, no handle found");
+  for (handle = handles; num_handles--; handle++)
+  {
+        if (d->handle == *handle)
+        {
+                found = 1;
+                break;
+        }
+  }
+  grub_free (handles);
+  if (! found)
+  {
+        return grub_error (GRUB_ERR_BAD_DEVICE, "device not exists");
+  }
+
   disk->id = ((num << GRUB_CHAR_BIT) | name[0]);
   m = d->block_io->media;
   /* FIXME: Probably it is better to store the block size in the disk,
